client/my-sites/stats/stats-period-navigation/index.jsx

```diff
@@ -213,7 +241,66 @@ class StatsPeriodNavigation extends PureComponent {
 	}
 }
 
-const connectComponent = connect( null, { recordGoogleEvent: recordGoogleEventAction } );
+const onGatedClick = ( siteId ) => ( shortcut, source ) => {
+	recordTracksEvent( `${ source }_stats_date_picker_shortcut_${ shortcut.id }_gated_clicked` );
+	recordTracksEvent( 'jetpack_stats_upsell_clicked', {
+		statType: shortcut.statType,
+		source,
+	} );
+	page( `/stats/purchase/${ siteId }?productType=personal&from=${ source }` );
+};
+
+const connectComponent = connect(
```

Comment:
Maybe something for a follow up but does this need to continue to use connect? Looks like it's only recordGoogleEvent which I think can be swapped out using something like `dispatch = useDispatch(); dispatch(recordGoogleEvent(...));`

Reply:
I think we'll still need to use connect because of the mapStateToProps part 🤔